woocommerce/woocommerce
# New - Implement remove_coupon method in WC_Order

### All Submissions:


### Changes proposed in this Pull Request:

This Pull Request adds a new remove_coupon method to the WC_Order class in WooCommerce. The remove_coupon method allows developers to remove a specific coupon from an order and recalculate the order totals. It also includes improvements for better readability, efficiency, and error handling.

The inclusion of parity for the woocommerce_removed_coupon hook ensures that developers can perform additional actions when a coupon is removed from an order.

Closes #36114.

### How to test the changes in this Pull Request:

1. Create a new order in WooCommerce that includes at least one coupon item.
2. Write a script that calls the remove_coupon method and passes in the coupon code of the coupon item to be removed.
3. Verify that the coupon item is successfully removed from the order and that the order totals are recalculated correctly.
4. Check that the usage count of the removed coupon is decreased by one.
5. Add an action hook to perform additional actions when a coupon is removed from an order and verify that the hook is triggered when the remove_coupon method is called.

### Other information:


### FOR PR REVIEWER ONLY:
